src/main/scala/viper/server/core/ViperCoreServer.scala

```diff
@@ -34,15 +34,16 @@ class ViperCoreServer(val config: ViperConfig)(implicit val executor: Verificati
     * This function must be called before any other. Calling any other function before this one
     * will result in an IllegalStateException.
     * */
-  def start(): Unit = {
+  def start(): Future[Unit] = {
     _logger = ViperLogger("ViperServerLogger", config.getLogFileWithGuarantee, config.logLevel())
     println(s"Writing [level:${config.logLevel()}] logs into " +
       s"${if (!config.logFile.isSupplied) "(default) " else ""}journal: ${logger.file.get}")
 
     ViperCache.initialize(logger.get, config.backendSpecificCache())
 
-    super.start(config.maximumActiveJobs())
-    logger.get.info(s"ViperCoreServer has started.")
+    super.start(config.maximumActiveJobs()) map { _ =>
+      logger.get.info(s"ViperCoreServer has started.")
+    }
```

Comment:
I'm surprised that this code type checks. If ```super.start``` returns a future, doesn't it get mapped to